runtime/ops/io.rs

```diff
@@ -301,33 +293,119 @@ impl Resource for ChildStderrResource {
   }
 }
 
+#[derive(Clone)]
+enum StdFileResourceInner {
+  // Ideally we would store stdio as an StdFile, but we get some Windows
+  // specific functionality for free by using Rust std's wrappers. So we
+  // take a bit of a complexity hit here in order to not have to duplicate
+  // the functionality in Rust's std/src/sys/windows/stdio.rs
+  Stdin,
+  Stdout,
+  Stderr,
+  File(Arc<Mutex<StdFile>>),
```

Comment:
Question: why do we store this in a mutex? Can't we just store `Arc<StdFile>` and clone it out when we need it or does this need exclusive access?

Reply:
I don't know off hand. I guess you'd have to remove it and see what happens.

Reply:
Discussed with Bert and the reason for this is because Read & Write both require `&mut` so storing this in a mutex is faster than duplicating the handle which also could fail.